packages/providers/src/provider.ts

```diff
@@ -316,9 +315,23 @@ export default class Provider {
         utxoValidation: false,
       });
       const receipts = gqlReceipts.map(processGqlReceipt);
-      missingOutputVariableCount = getReceiptsWithMissingOutputVariables(receipts).length;
+      const { missingOutputVariables, missingOutputContractIds } =
+        getReceiptsWithMissingData(receipts);
+
+      missingOutputVariableCount = missingOutputVariables.length;
+      missingOutputContractIdsCount = missingOutputContractIds.length;
+
+      if (missingOutputVariableCount === 0 && missingOutputContractIdsCount === 0) {
+        return;
+      }
+
       transactionRequest.addVariableOutputs(missingOutputVariableCount);
-    } while (tries > MAX_RETRIES || missingOutputVariableCount > 0);
+
+      missingOutputContractIds.forEach(({ contractId }) =>
+        transactionRequest.addContract(Address.fromString(contractId))
+      );
+      tries += 1;
+    } while (tries < MAX_RETRIES);
```

Comment:
What is the reasoning behind `MAX_RETRIES`, and why is it `10` and not `20` or `30` or other value?

Reply:
TBH, I just used the same number as the Rust SDK set.  Not sure what the motivation for it is too. cc @digorithm - should we set a max retry to a different number